p2p/src/net/libp2p/backend.rs

```diff
@@ -86,10 +92,33 @@ impl Backend {
         }
     }
 
+    /// Collect peers into a vector and send appropriate event to P2P
+    async fn send_discovery_event(
+        &mut self,
+        peers: Vec<(PeerId, Multiaddr)>,
+        discovered: bool,
```

Comment:
Not a massive fan of bool flags, as it's not very clear at the call site what is the meaning. Not sure what would be better short of a custom enum, which seems like an overkill. Here's one possible trick using a function instead of bool:
```rust
    event_fn: impl FnOnce(Vec<(PeerId, Multiaddr)>) -> common::Event,
(... blah blah ...)
    self.event_tx.send(event_fn(peers)).(...)
```
And then call it like so
```rust
    self.send_discovery_event(peers.collect(), common::Event::PeerDiscovered)
```

Reply:
Very nice, thanks. Fixed